apps/aecore/lib/aecore/chain/worker.ex

```diff
@@ -421,10 +440,11 @@ defmodule Aecore.Chain.Worker do
     state_update = %{
       state
       | blocks_data_map: hundred_blocks_data_map,
-        node_is_leader: new_node_is_leader
+        node_is_leader: new_node_is_leader,
+        generation_fees: new_generation_fees
     }
 
-    if top_height < height do
+    if top_height <= height && !block_is_genesis do
```

Comment:
I'm not sure about the Bitcoin-NG consensus rules, but this part of `:add_validated_block` handles changing top hash. I'm woried we might change to a fork with the same height or one of the previous microblocks.

To be sure we don't do that previously checking for increasing heights was enough. Now we have micro blocks with the same height. I think we could check now `top_height < height || (top_height == height && top_timestamp < timestamp)`

Reply:
Are we validating if the new block is linked to the genesis block?

Reply:
Isn't it impossible for it not to be linked to the genesis block if the genesis is always our first block and the next block has to be validated in pair with it?